src/utils.jl

```diff
@@ -97,3 +113,88 @@ function shuffle_split_stratified(
     end
     return inds1, inds2
 end
+
+_sample_dims(data::AbstractVector) = Colon()
+_sample_dims(data::AbstractArray{<:Any,3}) = (1, 2)
+
+"""
+    fold([f,] x::AbstractVector)
```

Comment:
IMO we should use a different name. It's too similar to foldl and foldr in Base, and IMO not clear enough.

Reply:
Agreed. Originally we also folded about the mean (the old approach for computing ESS of `std`), but now we only use it for `median`, so `_fold_median` may be better.